src/envoy/alts/tsi_transport_socket.cc

```diff
@@ -82,14 +85,23 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(
     tsi_handshaker_result_extract_peer(handshaker_result, &peer);
     ENVOY_CONN_LOG(debug, "TSI: Handshake successful: peer properties: {}",
                    callbacks_->connection(), peer.property_count);
-    // TODO(lizan): validate peers
     for (size_t i = 0; i < peer.property_count; ++i) {
       ENVOY_CONN_LOG(debug, "  {}: {}", callbacks_->connection(),
                      peer.properties[i].name,
                      std::string(peer.properties[i].value.data,
                                  peer.properties[i].value.length));
     }
+    std::string err = "";
+    bool peer_validated = handshake_validator_(peer, err);
```

Comment:
Do we need to check if handshake_validator_ is nullptr?

Reply:
I'm assuming it's used the same as handshake_factory which should always be provided by the caller. If the caller do not want to validate peer, they can pass in a lambda that always return true.

Reply:
After a second thought, I think it doesn't harm to add such check, so updated accordingly.